File: apps/api/src/Features/Movimiento/Models/MovimientoModel.cs
namespace Features.Movimiento.Models;

public class MovimientoModel
{
    public int IdMovimiento { get; set; }

    public int IdCuenta { get; set; }

    public string Tipo { get; set; } = string.Empty;

    public decimal Monto { get; set; }

    public string? Descripcion { get; set; }

    public int Mes { get; set; }

    public int Anio { get; set; }

    public string Via { get; set; } = string.Empty;

    public int? IdCuentaOrigen { get; set; }

    public int? IdCuentaDestino { get; set; }

    public DateTime Created { get; set; }

    public DateTime Updated { get; set; }
}

File: apps/web/src/app/features/movimiento/models/movimiento.model.ts
export type MovimientoTipo = 'acreditar' | 'debitar';

export type MovimientoVia = 'transferencia' | 'efectivo' | 'deposito' | 'retiro' | 'ajuste';

export interface Movimiento {
  idMovimiento: number;
  idCuenta: number;
  nombreCuenta?: string | null;
  tipo: MovimientoTipo;
  monto: number;
  descripcion?: string | null;
  via: MovimientoVia;
  idCuentaOrigen?: number | null;
  nombreCuentaOrigen?: string | null;
  idCuentaDestino?: number | null;
  nombreCuentaDestino?: string | null;
  created: string;
  updated: string;
}

export interface MovimientoCreate {
  idCuenta: number;
  tipo: MovimientoTipo;
  monto: number;
  descripcion?: string | null;
  via: MovimientoVia;
}

export interface MovimientoUpdate {
  idCuenta: number;
  tipo: MovimientoTipo;
  monto: number;
  descripcion?: string | null;
  via: MovimientoVia;
}

export interface MovimientoPorCuenta {
  idMovimiento: number;
  nombreCuenta: string;
  tipo: MovimientoTipo;
  monto: number;
  descripcion?: string | null;
  via: MovimientoVia;
  idCuentaOrigen?: number | null;
  nombreCuentaOrigen?: string | null;
  idCuentaDestino?: number | null;
  nombreCuentaDestino?: string | null;
  created: string;
  updated: string;
}

export interface TransferenciaCreate {
  idCuenta: number;
  idCuentaDestino: number;
  monto: number;
  descripcion?: string | null;
}